rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -833,17 +867,26 @@ public ResponseEntity<?> linkPackages(
             @Parameter(description = "Set of package IDs to be linked.",
                     example = "[\"3765276512\",\"5578999\",\"3765276513\"]"
             )
-            @RequestBody Set<String> packagesInRequestBody
-    ) throws URISyntaxException, TException {
-        if(!packageService.validatePackageIds(packagesInRequestBody)){
-            return new ResponseEntity<>("Package ID invalid! ", HttpStatus.NOT_FOUND);
-        }
-        RequestStatus linkPackageStatus = linkOrUnlinkPackages(id, packagesInRequestBody, true);
-        if (linkPackageStatus == RequestStatus.SENT_TO_MODERATOR) {
-            return new ResponseEntity<>(RESPONSE_BODY_FOR_MODERATION_REQUEST, HttpStatus.ACCEPTED);
-        }
-        return new ResponseEntity<>(HttpStatus.CREATED);
-    }
+            @RequestBody Set<String> packagesInRequestBody,
+            @Parameter(description = "Comment message.", example = "This is new MR.")
+            @RequestParam(value = "comment", required = false) String comment
+	) throws URISyntaxException, TException {
+		if (!packageService.validatePackageIds(packagesInRequestBody)) {
+			return new ResponseEntity<>("Package ID invalid! ", HttpStatus.NOT_FOUND);
+		}
+		final User sw360User = restControllerHelper.getSw360UserFromAuthentication();
+		Project project = projectService.getProjectForUserById(id, sw360User);
+		sw360User.setCommentMadeDuringModerationRequest(comment);
+		if (!restControllerHelper.isWriteActionAllowed(project, sw360User) && comment == null) {
+			return new ResponseEntity(RESPONSE_BODY_FOR_MODERATION_REQUEST_WITH_COMMIT, HttpStatus.ACCEPTED);
```

Comment:
In case of missing request parameter, shouldn't the response status be `BAD_REQUEST`?